File: exeercise-objects-references/src/Book.java
public class Book {
    String title;
    String author;
    boolean isAvailable = true;

    public Book(String title, String author) {
        this.title = title;
        this.author = author;
    }

    public void displayStatus() {
        System.out.printf("%s by %s available: %b%n", this.title, this.author, this.isAvailable);
    }

    public void borrowBook() {
        this.isAvailable = false;
    }
}

File: 05 - web dev practice/java-optional/exercise/book.js
const book = {
    title: "JavaScript Basics",
        author: {
            name: "John Doe",
            social: {
                twitter: "@johndoe"
        }
    },
    edition: {
        year: 1990,
    },
};

// Use optional chaining to access the Twitter handle
console.log(book.author?.social?.twitter);
// Try accessing a non-existent property using optional chaining
console.log(book.publisher?.name);

console.log(book.edition?.year)